llm/codellama/README.md

```diff
@@ -0,0 +1,185 @@
+CLOUD   INSTANCE                    vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE     COST ($)   CHOSEN   
+----------------------------------------------------------------------------------------------------------
+ Azure   Standard_NC48ads_A100_v4    48      440       A100-80GB:2    eastus          7.35          ✔     
+ GCP     g2-standard-96              96      384       L4:8           us-east4-a      7.98                
+ GCP     a2-ultragpu-2g              24      340       A100-80GB:2    us-central1-a   10.06               
+ Azure   Standard_NC96ads_A100_v4    96      880       A100-80GB:4    eastus          14.69               
+ GCP     a2-highgpu-4g               48      340       A100:4         us-central1-a   14.69               
+ AWS     g5.48xlarge                 192     768       A10G:8         us-east-1       16.29               
+ GCP     a2-ultragpu-4g              48      680       A100-80GB:4    us-central1-a   20.11               
+ Azure   Standard_ND96asr_v4         96      900       A100:8         eastus          27.20               
+ GCP     a2-highgpu-8g               96      680       A100:8         us-central1-a   29.39               
+ AWS     p4d.24xlarge                96      1152      A100:8         us-east-1       32.77               
+ Azure   Standard_ND96amsr_A100_v4   96      1924      A100-80GB:8    eastus          32.77               
+ GCP     a2-ultragpu-8g              96      1360      A100-80GB:8    us-central1-a   40.22               
+ AWS     p4de.24xlarge               96      1152      A100-80GB:8    us-east-1       40.97               
+----------------------------------------------------------------------------------------------------------
+
+Launching a cluster 'code-llama'. Proceed? [Y/n]: 
+```
+2. Send a request to the endpoint for code completion:
+```bash
+IP=$(sky status --ip code-llama)
+
+curl -L http://$IP:8000/v1/completions \
+    -H "Content-Type: application/json" \
+    -d '{
+      "model": "codellama/CodeLlama-70b-Python-hf",
+      "prompt": "def quick_sort(a: List[int]):",
+      "max_tokens": 256
+  }' | jq -r '.choices[0].text'
+```
+
+This returns the following completion:
+```python
+    if len(a) <= 1:
+        return a
+    pivot = a.pop(len(a)//2)
+    b = []
+    c = []
+    for i in a:
+        if i > pivot:
+            b.append(i)
+        else:
+            c.append(i)
+    b = quick_sort(b)
+    c = quick_sort(c)
+    res = []
+    res.extend(c)
+    res.append(pivot)
+    res.extend(b)
+    return res
+```
+
+## Scale up the service with SkyServe
+
+1. With [SkyServe](https://skypilot.readthedocs.io/en/latest/serving/sky-serve.html), scaling up the Code Llama service is as simple as running:
+```bash
+sky serve up -n code-llama ./endpoint.yaml
+```
+This will start the service with multiple replicas on the cheapest available locations and accelerators. SkyServe will automatically manage the replicas, monitor their health, and restart them when needed.
```

Comment:
```suggestion
This will start the service with multiple replicas on the cheapest available locations and accelerators. SkyServe will automatically manage the replicas, monitor their health, autoscale based on load, and restart them when needed.
```